File: Session_01_InputAndOutput/csharp/basics.cs
using System;

class Program {
    static void Main() {
        // 1. Output
        Console.WriteLine("Hello from C#!");

        // 2. Variables
        int age = 19;
        double gpa = 3.75;
        char grade = 'A';
        string name = "Budi";

        Console.WriteLine("Name: " + name);
        Console.WriteLine($"Age: {age} | GPA: {gpa} | Grade: {grade}");

        // 3. Input
        Console.Write("Enter a new age: ");
        string input = Console.ReadLine();
        age = Convert.ToInt32(input);

        Console.WriteLine("Updated Age: " + age);
    }
}

File: Session_01_InputAndOutput/rust/basics.rs
use std::io::{self, Write};

fn main() {
    // 1. Output
    println!("Hello from Rust!");

    // 2. Variables
    let mut age: i32 = 19;
    let gpa: f64 = 3.75;
    let grade: char = 'A';
    let name: &str = "Budi";

    println!("Name: {}", name);
    println!("Age: {} | GPA: {} | Grade: {}", age, gpa, grade);

    // 3. Input
    print!("Enter a new age: ");
    io::stdout().flush().unwrap(); // Flush output buffer so prompt displays immediately

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    
    // Parse input string to integer
    age = input.trim().parse().unwrap_or(age);

    println!("Updated Age: {}", age);
}
